students/MicahBraun/Lesson 04/mailroom.py

```diff
@@ -0,0 +1,290 @@
+#              with a nested list of stored information. User is able
+#              to add names to a list (not currently saved upon program
+#              exit), append values to fields, check against current
+#              fields, create mock "Thank You" e-mails based on user-
+#              -input and associated fields, and generate tabular display
+#              of data for viewing.
+# ------------------------------------------------------------------------
+import time                       # time to supply today's date for letters
+import os                         # for default working directory
+import locale                     # for currency formatting
+# Data -------------------------------------------------------------------
+locale.setlocale(locale.LC_ALL, '')
+today = time.strftime("%m/%d/%Y")
+spacing = '- ' * 55  # formatting for DONOR header
+data_list = [['NAMES', 'DONATION AMOUNT', 'NUMBER OF GIFTS', 'AVG. GIFTS'],
+             ['Rudolph Soares', 1500, 3, 1566],  # Donor info
+             ['Josef Mistretta', 250, 5, 235],
+             ['Joye Agostini', 5000, 2, 5750],
+             ['Joni Mattix', 2750, 1, 2750],
+             ['Rachelle Levan', 750, 3, 581],
+             ['Vena Ussery', 1000, 7, 785],
+             ['Efrain Lager', 10000, 1, 10000],
+             ['Mee Heine', 15000, 2, 9750],
+             ['Tanya Essex', 50000, 1, 50000],
+             ['Garrett Hartsell', 800, 2, 1400]]
+
+# Processing --------------------------------------------------------------
+
+def menu():
+    """Display menu of options to user"""
+    header = "-" * 100  # formatting for header
+    title = ' ' * 43 + "MAIL ROOM MENU"  # title display
+    directions = "Select one of the three options:"  # directions to user for menu
+    optA = "[A] Send a Thank you"  # option(s) for menu
+    optB = "[B] Create a Report"
+    optC = "[C] Create Letters for All Donors"
+    optD = "[D] Quit"
+    whole_menu_header = header + '\n' + title + '\n' + header  # header strung together as one uni
+
+    while True:
+        print(whole_menu_header)                                # display header
+        print(directions.center(100, ' '), '\n')                # center directions with header display
+        print(optA + optB.center(32, ' ') + optC.center(30, ' ') + optD.center(21, ' ') + '\n')
+
+        usr_sel = str(input("Menu Selection: ")).upper()        # 'select case' menu
+
+        if usr_sel == "A":
+            print("Leaving menu...")
+            thankyou()
+        elif usr_sel == "B":
+            print("Leaving menu...")
+            createreport()
+        elif usr_sel == "C":
+            print("Leaving menu...")
+            writeletters()
+        elif usr_sel == "D":
+            input("Exiting Program...[Press Enter]")
+            exit()
+        else:
+            # usr_sel != "A" or usr_sel != "B" or usr_sel != "C":
+            print("\nOnly enter A, B, C, or D!")
+            continue
```

Comment:
You should be using a [dict as switch](https://uwpce-pythoncert.github.io/PythonCertDevel/modules/DictionaryAsSwitch.html#dict-as-switch) for user selections per the Lesson 4 assignment instructions